### PR TITLE
Basics of C# Tutorial

### PR DESCRIPTION
**Pull request by danmo91**
Tutorial covers items in this [list](https://docs.google.com/document/d/1dL77H18Y1OAn4-Ezy0UVHyPpKjsLTu4Za9D3_n8j9Qw/edit?usp=sharing)

@profganderson this tutorial is to replace 
screen shot 2016-09-02 at 11 24 26 pm

let me know if it's okay

### CODE REVIEW

**Comment by danmo91** [comment]
@asmockler can you review and merge this?

File: basics-of-csharp/basics-of-csharp.md [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
+```
+
+Let's break down what each piece of code means.
+
+
+#### The Using Directive
```
Comment (asmockler) [inline basics-of-csharp/basics-of-csharp.md]: #### The `using` Directive

File: basics-of-csharp/basics-of-csharp.md [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+Data types help you define what kind of value a variable can contain.
+
+
+#### Primitive Types
+
+- bool (true/false)
+- byte (unsigned 8-bit integer range 0 to 255)
+- char ('a')
+- short, int, long
+- float, double, decimal
```
Comment (asmockler) [inline basics-of-csharp/basics-of-csharp.md]: - `bool` (true/false)
- `byte` (unsigned 8-bit integer range 0 to 255)
- `char` ('a')
- `short`, `int`, `long`
- `float`, `double`, `decimal`

File: basics-of-csharp/basics-of-csharp.md [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+    dan.Name = "Dan Morain";
+    dan.FavoritePlace = "Chamber of Secrets";
+  }
+}
+```
+
+
+#### Nullable
+
+Variables can not contain a value at all.  We call this `null`.
```
Comment (asmockler) [inline basics-of-csharp/basics-of-csharp.md]: Variables can contain no value at all. We call this `null`.

**Pull request merged**